src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -94,6 +99,45 @@ private Task OnProjectFactoryCompletedAsync()
 
             // Update project context and subscriptions.
             await UpdateProjectContextAndSubscriptionsAsync().ConfigureAwait(false);
+            LogLanguageServiceHostInitializeStop();
+        }
+
+        private static volatile bool s_isFirstStartEvent = true;
+        private static volatile int s_numberOfTimesInitialized = 0;
+        private static object s_telemetryLock = new object();
+
+        /// <summary>
+        /// Only posts an event the first time this method is called in the VS process
+        /// Also updates a global count so we know how many project hosts have been initialized
+        /// </summary>
+        private void LogLanguageServiceHostInitializeStart()
+        {
+            if (s_isFirstStartEvent)
+            {
+                lock (s_telemetryLock)
+                {
+                    if (s_isFirstStartEvent)
+                    {
+                        s_isFirstStartEvent = false;
+                        _telemetryService.PostEvent("LanguageServiceHost/Start");
+                    }
+                }
+            }
+
+            Interlocked.Increment(ref s_numberOfTimesInitialized);
+        }
+
+        /// <summary>
+        /// Decrements the global count and reports that all hosts have been initialized
+        /// if the count is zero.
+        /// </summary>
+        private void LogLanguageServiceHostInitializeStop()
+        {
+            Interlocked.Decrement(ref s_numberOfTimesInitialized);
+            if (s_numberOfTimesInitialized == 0)
+            {
+                _telemetryService.PostEvent("LanguageServiceHost/Stop");
```

Comment:
I'm not *sure* that this sequence is guaranteed to work.  I could imagine it being possible that we could start these and stop them in an interleaved fashion allowing things to get to zero multiple times?

@davkean @lifengl - can you comment on if there are any guarantees here?

Reply:
No guarantees that these don't interleave.

Reply:
I'd prefer; 1 not including the type name in the event. This entire class has been removed in my current change there is no LanguageServiceHost.

Reply:
@Pilchie for solution open I am not seeing any interleaving, though I agree it is possible.  If we get multiple stop events in traces that would invalidate the data.